From 4cd2e15301c9fd237b295c3a7e8f9f6ab8052cfa Mon Sep 17 00:00:00 2001
From: chenyutao <chenyutao2@huawei.com>
Date: Tue, 10 Mar 2026 21:51:59 +0800
Subject: [PATCH] urma: check ioctl errno value

This patch is for checking ioctl errno value,
when the value equals URMA_KERNEL_DRV_ERRNO, a
kernel driver error is especailly exposed.

Signed-off-by: chenyutao <chenyutao2@huawei.com>
(cherry picked from commit 540dab5f41f6bedac8075d901fa12829a6cfba25)
---
 src/urma/lib/urma/core/urma_cmd_tlv.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/src/urma/lib/urma/core/urma_cmd_tlv.c b/src/urma/lib/urma/core/urma_cmd_tlv.c
index 312924d4..e918e942 100644
--- a/src/urma/lib/urma/core/urma_cmd_tlv.c
+++ b/src/urma/lib/urma/core/urma_cmd_tlv.c
@@ -13,6 +13,9 @@
 
 #include "urma_cmd_tlv.h"
 
+/* between 1 and MAX_ERRNO */
+#define URMA_KERNEL_DRV_ERRNO 2048
+
 static inline void fill_attr(urma_cmd_attr_t *attr, uint16_t type, uint16_t field_size, uint16_t el_num,
                              uint16_t el_size, uintptr_t data)
 {
@@ -48,7 +51,8 @@ static int urma_tlv_ioctl(int ioctl_fd, urma_cmd_t cmd, urma_cmd_attr_t *args, u
      };
     int ret = ioctl(ioctl_fd, URMA_CMD, &hdr);
     if (ret != 0) {
-        URMA_LOG_ERR("ioctl failed, ret:%d, errno:%d, cmd:%u.\n", ret, errno, hdr.command);
+        URMA_LOG_ERR("ioctl failed, ret:%d, errno:%d, cmd:%u, kdrv_err: %d.\n",
+            ret, errno, hdr.command, (int)(errno == URMA_KERNEL_DRV_ERRNO));
     }
     return ret;
 }
-- 
GitCode

